client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -339,8 +345,9 @@ private int calculateNumSegmentsToAcquire(ReaderGroupState state) {
         return Math.max(Math.max(equallyDistributed, fairlyDistributed), 1);
     }
 
-    private Duration calculateAcquireTime(ReaderGroupState state) {
-        return TIME_UNIT.multipliedBy(state.getNumberOfReaders() - state.getRanking(readerId));
+    @VisibleForTesting
+    static Duration calculateAcquireTime(String readerId, ReaderGroupState state) {
+        return TIME_UNIT.multipliedBy(2 * (state.getNumberOfReaders() - state.getRanking(readerId)));
```

Comment:
I don't understand why the `2` factor in this computation.

Reply:
That was a hack. I have replaced it with a new check that will prevent 'full' hosts from being over-eager if they haven't acquired for a while because checkpoints are infrequent.